src/QueryBuilder.php

```diff
@@ -177,12 +160,30 @@ public function get() : Collection
         return collect($posts);
     }
 
+    public function paginate($perPage = 10, $page = 1): PostQuery
+    {
+        global $paged;
+
+        if (isset($page)) {
```

Comment:
I was missing this feature and stumbled upon this merge request. I quickly checked how it was implemented by curiosity and I might have found an issue here : )

Since the `$page` attribute has a default value in the function params it will always be set. This means that if the user doesn't pass the $page attribute this will always reset the `$paged` value to 1.

One solution is to set the `$page = NULL` in the params and check if `$page !== NULL` here instead of the isset. I still didn't test it in a real wordpress setup so I might be wrong.

Reply:
Thanks for the feedback, I guess the `if` is redundant here. I can't see that it would cause any issue but you're right that it will always be set.

I think it's important to provide a default as I can't think of a scenario where when providing no `$page` param you wouldn't want the first page back.